rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp

```diff
@@ -75,6 +75,34 @@ align_int_(size_t __align, T __int) noexcept
   return (__int - 1u + __align) & ~(__align - 1);
 }
 
+template<typename MembersType>
+void resize_field(
+  const MembersType * member,
+  void * field,
+  size_t size)
+{
+  if (!member->resize_function) {
+    throw std::runtime_error("unexpected error: resize function is null");
```

Comment:
if we're throwing here, we have to make sure to catch this exception everywhere else, as we cannot throw an exception through the cpp/c "boundary"

Reply:
I agree with that. Every invocation of `deserializeROSmessage` in a C API implementation is in a try-catch block (see [here](https://github.com/ros2/rmw_cyclonedds/blob/8857f2ddead5bfc84d64207bc72bf75c1928a5cd/rmw_cyclonedds_cpp/src/rmw_node.cpp#L1459) and [here](https://github.com/ros2/rmw_cyclonedds/blob/8857f2ddead5bfc84d64207bc72bf75c1928a5cd/rmw_cyclonedds_cpp/src/rmw_node.cpp#L1459)), besides checking for falsy return values. Any new code that uses these APIs has to be careful about it.

Reply:
Those two snippets were the same :sweat_smile:. But you're right. One isn't.

Reply:
See https://github.com/ros2/rmw_cyclonedds/pull/313/commits/694364f3fea275b7c99bef04f5de250f55606608.

Reply:
> Those two snippets were the same

hahaha, I checked both like if they were different :joy: